slack-import: Reorder and extract additional helper functions in `slack_message_conversion`

PieterCK opened the pull request:
This is a prep PR for making the Slack text reformatters in `slack_message_conversion.py` to be compatible with `render_attachments` & `render_blocks` (#32164), it refactors the `slack_message_conversion.py` file. Changes include:
- Rearranging helper functions to appear before the main functions that use them.
- Create `convert_slack_workspace_mentions` helper function.
- Create `convert_slack_formatting` helper function.

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [ ] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [ ] Each commit is a coherent idea.
- [ ] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [ ] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

laurynmm commented:
@PieterCK - Thanks for breaking these refactors out to a separate PR and commits that are very clear to review! I made a few small updates, noted below, so that I can go ahead and mark this for integration review.

**Updates**:
- Squashed the commits rearranging functions. It was very easy to review that these were moving the functions in the file without any changes, but in the end can be one commit in the git history.
- Updated the comment for `convert_slack_workspace_mentions` to be a bit more readable.
- Removed an empty line that was accidentally added to the test file in the final commit.

PieterCK commented:
@laurynmm I accidentally overwrote your changes :sweat:. I've manually added back the updates you mentioned in your comment. Do you mind taking a quick look to make sure I didn’t miss anything?
- Squashed the rearranging work to this commit [2fe0e9f](https://github.com/zulip/zulip/pull/32430/commits/2fe0e9ff27bc3bbd3a999d14f5444b79fae2bc14)
- Added back the new code comment for `convert_slack_workspace_mentions`
Other than that, I've moved the changes for `replace_links` to the main PR in #32164 as per this [comment](https://github.com/zulip/zulip/pull/32164#discussion_r1870212169).

Sorry!

laurynmm commented:
@PieterCK - No worries, I imagine it just came from the feedback you were getting from the related PR. I think the only thing you missed were some updates to the commit messages I made. Once you've reviewed and updated for your preferences there, as you're the author of these changes, let me know and I'll do one more pass before marking for integration review.

It'd be good to get these changes that don't change how the code works off of the more complicated regex changes in #32124. I might update the title of this PR too, maybe "slack-import: Reorder and extract additional helper functions in slack_message_conversion".

commit 1:
```
slack_import: Reorder functions used in convert_to_zulip_markdown.

This prep commit moves `convert_to_zulip_markdown` and
`convert_markdown_syntax` so that they appear after the various
helper functions used in the main logic for converting Slack
messages to Zulip's formatting. This adjustment aligns with code
style conventions, improving readability.

The change also prepares the codebase for reusing Slack text
reformatting logic in the Slack incoming webhook.
```

commit 2:
```
slack_import: Create and use convert_slack_workspace_mentions helper.

This prep commit extracts the logic for reformatting Slack mentions into
Zulip mentions from `convert_to_zulip_markdown` into a new helper
function, `convert_slack_workspace_mentions`.

This is done to make the reformatting logic be reusable else where such
as in the Slack webhook or Slack incoming webhook.
```

commit 3:
```
slack_import: Create and use convert_slack_formatting helper.

This prep commit extracts the logic for converting Slack formatted text
into Zulip friendly markdown from `convert_to_zulip_markdown` into a new
helper function, `convert_slack_formatting`.

This is done to make reformating logic be reusable else where such as in
the Slack webhook or Slack incoming webhook.
```

PieterCK commented:
Updated the commit description, this is ready for another look through
